### ISSUE TITLE
panic when importing an S3 bucket

### ISSUE THREAD

**Issue by sunbing81** [pulumi/pulumi#5860]
`pulumi import 'aws:s3/bucket:Bucket' name id` just threw a panic:

```
Previewing import (shared-resources-and-services):
     Type                 Name                                                         Plan
     pulumi:pulumi:Stack  shared-resources-and-services-shared-resources-and-services  running
 =   └─ aws:s3:Bucket     name                                                         import
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xe7c776]

goroutine 58 [running]:
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*deploymentExecutor).importResources(0xc00133ed60, 0x2003340, 0xc00002e420, 0x7f0f1b50acd0, 0xc0001ae280, 0x7fffffff, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/runner/work/pulumi/pulumi/pkg/resource/deploy/deployment_executor.go:459 +0x276
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*deploymentExecutor).Execute(0xc00133ed60, 0x2003340, 0xc00002e420, 0x7f0f1b50acd0, 0xc0001ae280, 0x7fffffff, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/runner/work/pulumi/pulumi/pkg/resource/deploy/deployment_executor.go:138 +0x1247
github.com/pulumi/pulumi/pkg/v2/resource/deploy.(*Deployment).Execute(...)
        /Users/runner/work/pulumi/pulumi/pkg/resource/deploy/deployment.go:383
github.com/pulumi/pulumi/pkg/v2/engine.(*deployment).run.func1(0x200ddc0, 0xc0001ae280, 0xc001ae6a00, 0x2003340, 0xc00002e420, 0xc00002e401, 0xc00136d480, 0xc0005fad20)
        /Users/runner/work/pulumi/pulumi/pkg/engine/deployment.go:251 +0x267
created by github.com/pulumi/pulumi/pkg/v2/engine.(*deployment).run
        /Users/runner/work/pulumi/pulumi/pkg/engine/deployment.go:237 +0x306
```

`pulumi version`: v2.15.0

`pulumi plugin ls`:  aws   resource  3.16.0

**Comment by stack72** [comment]
Hi @sunbing81 

My apologies you are facing this issue - can you give me a little more info here - what is language of the Pulumi project you are trying to import the resource into?

Paul

**Comment by sunbing81** [comment]
Hi, sorry for the inconvenience.

Both typescript and python projects hit into the issue. I tested some examples (e.g. a static S3 website) for the typescript project and everything is fine, except the import.

Thanks.

**Comment by timmyers** [comment]
I am also seeing this trying to `pulumi import aws:elasticache/cluster:Cluster name id` with a typescript program.
```
❯ pulumi version
v2.15.1
```

**Comment by timmyers** [comment]
I think I found the issue trying to do the `import` the old way.  This is happening when you would otherwise get this error:

```
Previewing update (infrastructure.legacy):
     Type                        Name                                  Plan       Info
     pulumi:pulumi:Stack         infrastructure-infrastructure.legacy             1 error
 =   └─ aws:elasticache:Cluster  head-tracker                          import     1 error
 
Diagnostics:
  pulumi:pulumi:Stack (infrastructure-infrastructure.legacy):
    error: preview failed
 
  aws:elasticache:Cluster (head-tracker):
    error: Preview failed: 1 error occurred:
        * missing required configuration key "aws:region": The region where AWS operations will take place. Examples are us-east-1, us-west-2, etc.
    Set a value using the command `pulumi config set aws:region <value>`.
```

**Comment by stack72** [comment]
Hi all

I have just released Pulumi v2.15.2 - please can you let me know if this still occurs in v2.15.2

Paul

**Comment by timmyers** [comment]
I see the same issue with `2.15.2`

**Comment by timmyers** [comment]
I also see basically the same error if the `id` does not actually exist in the cloud.

**Comment by stack72** [comment]
Hi @sunbing81  / @timmyers 

My apologies you are seeing this issue - I just opened a PR that fixes the problem and it will be made available in Pulumi v2.15.4 later today

Paul
